bin/mailmap_check.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+A tool to generate AUTHORS. We started tracking authors before moving to git,
+so we have to do some manual rearrangement of the git history authors in order
+to get the order in AUTHORS. bin/mailmap_check.py should be run before
+committing the results.
+
+See here for instructions on using this script:
+https://github.com/sympy/sympy/wiki/Development-workflow
+"""
+
+from __future__ import unicode_literals
+from __future__ import print_function
+
+import sys
+import os
+from pathlib import Path
+from subprocess import run, PIPE
+from collections import OrderedDict, defaultdict
+from argparse import ArgumentParser
+
+if sys.version_info < (3, 7):
+    sys.exit("This script requires Python 3.7 or newer")
+
+def sympy_dir():
+    return Path(__file__).resolve().parent.parent
+
+# put sympy on the path
+sys.path.insert(0, str(sympy_dir()))
+import sympy
+from sympy.utilities.misc import filldedent
+from sympy.external.importtools import version_tuple
+
+
+def main(*args):
+
+    parser = ArgumentParser(description='Update the .mailmap and/or AUTHORS files')
+    parser.add_argument('--update-authors', action='store_true',
+            help=filldedent("""Also update the AUTHORS file. Note that it
+            should only necessary for the release manager to do this as part of
+            the release process for SymPy."""))
+    args = parser.parse_args(args)
+
+    if not check_git_version():
+        return 1
+
+    # find who git knows ahout
+    try:
+        git_people = get_authors_from_git()
+    except AssertionError as msg:
+        print(red(msg))
+        return 1
+
+    lines_mailmap = read_lines(mailmap_path())
+
+    def key(line):
+        # return lower case first address on line or
+        # raise an error if not an entry
+        if '#' in line:
+            line = line.split('#')[0]
+        L, R = line.count("<"), line.count(">")
+        assert L == R and L in (1, 2)
+        return line.split(">", 1)[0].split("<")[1].lower()
+
+    who = OrderedDict()
+    for i, line in enumerate(lines_mailmap):
+        try:
+            who.setdefault(key(line), []).append(line)
+        except AssertionError:
+            who[i] = [line]
+
+    problems = False
+    missing = False
+    ambiguous = False
+    dups = defaultdict(list)
+
+    for person in git_people:
+        email = key(person)
+        dups[email].append(person)
+        if email not in who:
+            print(red("This author is not included in the .mailmap file:"""))
```

Comment:
The extra quotes at the end here are really confusing my text editor

```suggestion
            print(red("This author is not included in the .mailmap file:"))
```